controllers/medusa/medusaconfiguration_controller.go

```diff
@@ -0,0 +1,121 @@
+/*
+Copyright 2022.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package medusa
+
+import (
+	"context"
+	"github.com/go-logr/logr"
+	"github.com/k8ssandra/k8ssandra-operator/pkg/config"
+	corev1 "k8s.io/api/core/v1"
+	"k8s.io/apimachinery/pkg/api/errors"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/types"
+
+	"k8s.io/apimachinery/pkg/runtime"
+	ctrl "sigs.k8s.io/controller-runtime"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+	"sigs.k8s.io/controller-runtime/pkg/log"
+
+	medusav1alpha1 "github.com/k8ssandra/k8ssandra-operator/apis/medusa/v1alpha1"
+)
+
+// MedusaConfigurationReconciler reconciles a MedusaConfiguration object
+type MedusaConfigurationReconciler struct {
+	*config.ReconcilerConfig
+	client.Client
+	Scheme *runtime.Scheme
+}
+
+//+kubebuilder:rbac:groups=medusa.k8ssandra.io,namespace="k8ssandra",resources=medusaconfigurations,verbs=get;list;watch;create;update;patch;delete
+//+kubebuilder:rbac:groups=medusa.k8ssandra.io,namespace="k8ssandra",resources=medusaconfigurations/status,verbs=get;update;patch
+//+kubebuilder:rbac:groups=medusa.k8ssandra.io,namespace="k8ssandra",resources=medusaconfigurations/finalizers,verbs=update
+
+// Reconcile is part of the main kubernetes reconciliation loop which aims to
+// move the current state of the cluster closer to the desired state.
+// TODO(user): Modify the Reconcile function to compare the state specified by
+// the MedusaConfiguration object against the actual cluster state, and then
+// perform operations to make the cluster state reflect the state specified by
+// the user.
+//
+// For more details, check Reconcile and its Result here:
+// - https://pkg.go.dev/sigs.k8s.io/controller-runtime@v0.14.1/pkg/reconcile
+func (r *MedusaConfigurationReconciler) Reconcile(ctx context.Context, req ctrl.Request) (ctrl.Result, error) {
+	logger := log.FromContext(ctx).WithValues("medusabackupjob", req.NamespacedName)
+
+	logger.Info("Starting reconciliation")
+
+	// Fetch the MedusaConfiguration instance
+	instance := &medusav1alpha1.MedusaConfiguration{}
+	err := r.Get(ctx, req.NamespacedName, instance)
+	if err != nil {
+		logger.Error(err, "Failed to get MedusaConfiguration")
+		if errors.IsNotFound(err) {
+			return ctrl.Result{}, nil
+		}
+		return ctrl.Result{}, err
+	}
+
+	configuration := instance.DeepCopy()
+	patch := client.MergeFrom(configuration.DeepCopy())
```

Comment:
[question] why call `DeepCopy` twice?

Reply:
The patch and the original object we're patching need to be in separate addresses for when we invoke `.Status().Patch()`.
I'm not sure why we initially do a DeepCopy() on the object we get from the kube api, but I'm just following the convention we use elsewhere, which could be unnecessary but at least works fine :)